docs/time_series_causal_graph.md

```diff
@@ -0,0 +1,349 @@
+
+# Get the minimal graph
+minimal_graph = ts_cg.get_minimal_graph()
+
+# Check whether a graph is in its minimal form
+is_minimal = ts_cg.is_minimal_graph()
+```
+
+## Summary graph
+You can collapse the graph in time into a single node per variable (column name).
+This can become cyclic and bi-directed as X(t-1) -> Y and Y(t-1) -> X would become X <-> Y.
+Note that the summary graph is a `CausalGraph` object.
+
+```python
+from cai_causal_graph import TimeSeriesCausalGraph
+
+ts_cg: TimeSeriesCausalGraph
+
+summary_graph = ts_cg.get_summary_graph()
+```
+
+## Example
+Time series causal graph.
+X1(t-2)-> X1(t-1) -> X2(t-1) -> X2(t), X1(t) -> X2(t), X2(t-2) -> X2(t-1)
+![ts_cg](images/ts_cg.png)
+
+Minimal graph.
+X1(t-1) -> X1(t) -> X2(t), X2(t-1) -> X2
+![ts_minimal_graph](images/ts_minimal_graph.png)
+
+Summary graph.
+X1 -> X2
+![ts_summary_graph](images/ts_summary_graph.png)
+
+
+## Extended graph
+You can extend the graph in time by adding nodes for each variable at each time step from `backward_steps` to
+`forward_steps`. If a backward step of n is specified, it means that the graph will be extended in order to
+include nodes back to time -n and all the nodes connected to them as specified by the minimal graph.
+If both `backward_steps` and `forward_steps` are None, the original graph is returned.
+
+```python
+from cai_causal_graph import TimeSeriesCausalGraph
+
+ts_cg: TimeSeriesCausalGraph
+backward_steps = 2
+forward_steps = 3
+
+extended_graph = ts_cg.extend_graph()
+```
+
+## Query the nodes and the variables
+You can extract all the variable names from the variable nodes of the time series graph.
+
+```python
+from cai_causal_graph import TimeSeriesCausalGraph
+
+ts_cg: TimeSeriesCausalGraph
+
+# Return a list of variable names from a list of node names
+node_names = ['X', 'X lag(n=1)', 'Y', 'Z lag(n=2)']
+ts_cg.get_variable_names_from_node_names()
+>>> ['X', 'Y', 'Z']
+```
+
+Conversely, you can extract the time series nodes in the graph from a given list of identifiers.
+
+```python
+from cai_causal_graph import TimeSeriesCausalGraph
+
+ts_cg: TimeSeriesCausalGraph
+
+# A single node
+variable = ts_cg.get_nodes(identifier='X lag(n=1)')
+
+# Multiple nodes
+variable = ts_cg.get_nodes(identifier=['X', 'X lag(n=1)'])
+```
+
+## Add nodes
+You can add time series nodes and the corresponding meta data will be automatically populated.
+
+```python
+from cai_causal_graph import TimeSeriesCausalGraph, TimeSeriesNode
+
+ts_cg: TimeSeriesCausalGraph
+
+# Via a node object
+new_node: TimeSeriesNode
+ts_cg.add_node(new_node)
+# Via identifier
+ts_cg.add_node(identifier='X lag(n=3)')
+# Via variable name and time lag
+ts_cg.add_node(variable_name='X', time_lag=3)
+
+# Variable type can also be specified
+ts_cg.add_node(identifier='X lag(n=3)', variable_type=NodeVariableType.UNSPECIFIED)
+
+# You can also provide the additional meta information
+ts_cg.add_node(identifier='X lag(n=3)', meta={TIME_LAG: 3, VARIABLE_NAME: 'X'})
```

Comment:
I would not do this actually.  We fill this automatically from the identifier. Would hide how we use meta to keep this info away from the user.

Reply:
This was not resolved but marked as resolved. Reopening.